mapper-runtime/src/main/java/com/datastax/oss/driver/internal/mapper/DefaultMapperContext.java

```diff
@@ -73,7 +92,19 @@ public DefaultMapperContext withKeyspaceAndTable(
             && Objects.equals(newTableId, this.tableId))
         ? this
         : new DefaultMapperContext(
-            session, newKeyspaceId, newTableId, nameConverterCache, customState);
+            session, newKeyspaceId, newTableId, null, nameConverterCache, customState);
+  }
+
+  public DefaultMapperContext withExecutionProfile(@Nullable String newExecutionProfile) {
+    return newExecutionProfile.equals(this.executionProfile)
+        ? this
+        : new DefaultMapperContext(
+            session, keyspaceId, tableId, newExecutionProfile, nameConverterCache, customState);
+  }
+
+  public DefaultMapperContext withExecutionProfile(
+      @Nullable DriverExecutionProfile newExecutionProfile) {
+    return withExecutionProfile(newExecutionProfile.getName());
```

Comment:
This won't work with [derived profiles](https://docs.datastax.com/en/developer/java-driver/4.5/manual/core/configuration/#derived-profiles): the name references the profile they were copied from, it won't reflect any option that was changed dynamically.

We need a separate `DriverExecutionProfile` field in the context. It's the same as statement types in the core driver.